mock-infrastructure/src/main/java/com/sequenceiq/mock/verification/RequestResponseStorageService.java

```diff
@@ -1,16 +1,18 @@
 package com.sequenceiq.mock.verification;
 
 import java.util.ArrayList;
-import java.util.HashMap;
+import java.util.Collections;
 import java.util.List;
 import java.util.Map;
+import java.util.concurrent.ConcurrentHashMap;
+import java.util.stream.Collectors;
 
 import org.springframework.stereotype.Component;
 
 @Component
 public class RequestResponseStorageService {
 
-    private Map<String, List<Call>> calls = new HashMap<>();
+    private Map<String, List<Call>> calls = new ConcurrentHashMap<>();
 
     public void put(String testName, Call call) {
         List<Call> calls = this.calls.computeIfAbsent(testName, key -> new ArrayList<>());
```

Comment:
Can you please use a thread safe `CopyOnWriteArrayList` instead of the `ArrayList`?

Reply:
done